homeassistant/components/honeywell/climate.py

```diff
@@ -266,42 +280,44 @@ async def _set_temperature(self, **kwargs) -> None:
             # Get current mode
             mode = self._device.system_mode
             # Set hold if this is not the case
-            if getattr(self._device, f"hold_{mode}", None) is False:
-                # Get next period key
-                next_period_key = f"{mode.capitalize()}NextPeriod"
-                # Get next period raw value
-                next_period = self._device.raw_ui_data.get(next_period_key)
+            if self._device.hold_heat is False and self._device.hold_cool is False:
                 # Get next period time
-                hour, minute = divmod(next_period * 15, 60)
+                hour_heat, minute_heat = divmod(
+                    self._device.raw_ui_data.get("HEATNextPeriod", 0) * 15, 60
+                )
+                hour_cool, minute_cool = divmod(
+                    self._device.raw_ui_data.get("COOLNextPeriod", 0) * 15, 60
+                )
                 # Set hold time
                 if mode in COOLING_MODES:
-                    await self._device.set_hold_cool(datetime.time(hour, minute))
-                elif mode in HEATING_MODES:
-                    await self._device.set_hold_heat(datetime.time(hour, minute))
-
-            # Set temperature
-            if mode in COOLING_MODES:
-                await self._device.set_setpoint_cool(temperature)
-            elif mode in HEATING_MODES:
-                await self._device.set_setpoint_heat(temperature)
+                    await self._device.set_hold_cool(
+                        datetime.time(hour_cool, minute_cool)
+                    )
+                if mode in HEATING_MODES:
+                    await self._device.set_hold_heat(
+                        datetime.time(hour_heat, minute_heat)
+                    )
+
+            # Set temperature if not in auto
+            else:
+                if mode == "cool":
+                    await self._device.set_setpoint_cool(temperature)
+                if mode == "heat":
+                    await self._device.set_setpoint_heat(temperature)
+                if mode == "auto":
+                    if temperature := kwargs.get(ATTR_TARGET_TEMP_HIGH):
+                        await self._device.set_setpoint_cool(temperature)
+                    if temperature := kwargs.get(ATTR_TARGET_TEMP_LOW):
+                        await self._device.set_setpoint_heat(temperature)
```

Comment:
All these if-statements should be merged:

```suggestion

            # Set temperature if not in auto
            elif mode == "cool":
                await self._device.set_setpoint_cool(temperature)
            elif mode == "heat":
                await self._device.set_setpoint_heat(temperature)
            elif mode == "auto" and temperature := kwargs.get(ATTR_TARGET_TEMP_HIGH):
                await self._device.set_setpoint_cool(temperature)
            elif temperature := kwargs.get(ATTR_TARGET_TEMP_LOW):
                await self._device.set_setpoint_heat(temperature)
```

Reply:
Yes, got a bit carried away removing `elif` that was causing the problem